### PR TITLE
Pause contracts

### PR DESCRIPTION
**Pull request by smbsp**
Implementation Details: https://docs.google.com/document/d/1Qe92Q_1t3Ajwvurd9_8RtV8RMGaq30hNyij9_g2QWYo/edit

Needs more testing.

closes: #210

### CODE REVIEW

File: contracts/interfaces/ISovryn.sol [hunk 1]
```diff
@@ -111,6 +111,10 @@ contract ISovryn is
 
 	function setRebatePercent(uint256 rebatePercent) external;
 
+	function pause() external;
+
+	function unpause() external;
```
Comment (tjcloa) [inline contracts/interfaces/ISovryn.sol:116]: why not just one function toggleProtocolPause() ?

**Comment by tjcloa** [comment]
here is an overall idea of implementation
as discussed primarily with @smbsp 
1. `_setTarget(this.swapExternal.selector, target); `
is replaced for 
`_setTarget(this.swapExternal.selector>>1, target);`
in all modules
that will lead to calling non-existing functions from the protocol, but this is controllable (see next)
2. add fallback functions to all modules which will serve as a hook prior to calling any module function
3. in the fallback function 
- check that the protocol is not on pause and virtually any preliminary processing can be done there
- unshift func sig to get the real one: msg.sig<<1 
- check that the unshifted (msg.sig<<1) sig exists
- call the func with unshifted (msg.sig<<1) signature

File: contracts/interfaces/ISovryn.sol (hunk 1 above)
Comment (smbsp) [reply contracts/interfaces/ISovryn.sol]: Removed Pause/Unpause and created a new function toggleFreeze(). Added test cases for one more module - LoanSettings

**Review by korepkorep — COMMENTED** [review]
Several code quality issues that don't endanger contracts` security

File: contracts/modules/ProtocolSettings.sol [hunk 2]
```diff
@@ -593,4 +595,9 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents {
 	function getLockedSOVAddress() external view returns (address) {
 		return lockedSOVAddress;
 	}
+
+	function togglePaused(bool paused) external onlyOwner {
+		pause = paused;
```
Comment (denis-abag) [inline contracts/modules/ProtocolSettings.sol:600]: require the pause value is changing

File: contracts/modules/ProtocolSettings.sol [hunk 3]
```diff
@@ -593,4 +595,9 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents {
...
+		emit TogglePaused(msg.sender, !paused, paused);
```
[the remark this answers is not in the document]
Comment (smbsp) [reply contracts/modules/ProtocolSettings.sol:602]: Added a check to check if the parameter differs

**Review by tjcloa — APPROVED** [review]

**Pull request merged**